addons/stock_account/stock.py

```diff
@@ -297,7 +300,10 @@ def _invoice_create_line(self, cr, uid, moves, journal_id, inv_type='out_invoice
             else:
                 invoice = invoice_obj.browse(cr, uid, invoices[key], context=context)
                 if not invoice.origin or invoice_vals['origin'] not in invoice.origin.split(', '):
-                    invoice_origin = filter(None, [invoice.origin, invoice_vals['origin']])
+                    list1 = invoice.origin.split(' ,')
+                    list2 = invoice_vals['origin'].split(' ,')
+                    invoice_origin = list1 + list(set(list2) - set(list1))
+                    invoice_origin = filter(None, invoice_origin)
```

Comment:
Your code seems quite complicated. Are you trying to do something like
elements = (invoice.origin or '').split(',') + (invoice_vals['origin'] or '').split(',')
invoice_origin = ', '.join(set(e.strip() for e in elements)) ?